### PR TITLE
Update theoretical description and fix bugs in classification tutorial

### LINKED ISSUE #174: Theoretical description for classification missing
**Is your documentation request related to a problem? Please describe.**
The theoretical description for classification is still missing.

**Describe the solution you'd like**
The description of the three methods implemented so far in `MapieClassifier` : LABEL, APS, and Top-K.

### PR DESCRIPTION
**Pull request by vtaquet**
# Description

Add theoretical description for classification and fix a few bugs on classification tutorial.

Fixes #174 

## Type of change

Please remove options that are irrelevant.

- This change requires a documentation update

# Checklist

- [x] I have read the [contributing guidelines](https://github.com/simai-ml/MAPIE/blob/master/CONTRIBUTING.rst)
- [x] I have updated the [HISTORY.rst](https://github.com/simai-ml/MAPIE/blob/master/HISTORY.rst) and [AUTHORS.rst](https://github.com/simai-ml/MAPIE/blob/master/AUTHORS.rst) files
- [x] Linting passes successfully : `make lint`
- [x] Typing passes successfully : `make type-check`
- [x] Unit tests pass successfully : `make tests`
- [x] Coverage is 100% : `make coverage`
- [x] Documentation builds successfully : `make doc`

### CODE REVIEW

File: doc/theoretical_description_classification.rst [hunk 1]
```diff
@@ -6,15 +6,179 @@
 Theoretical Description
```
[the remark this answers is not in the document]
Comment (LacombeLouis) [reply doc/theoretical_description_classification.rst:6]: --> wil check all the equations once documentation is built

File: doc/theoretical_description_classification.rst [hunk 2]
```diff
@@ -6,15 +6,179 @@
...
+The so-called Adaptive Prediction Set (APS) method overcomes the problem encountered by the LABEL method through the construction of
+prediction sets which are by definition non-empty.
+The conformity scores are computed by summing the ranked scores of each label, from the higher to the lower until reaching the true
+label of the observation :
+
+.. math:: 
+   s_i(X_i, Y_i) = \sum^k_{j=1} \hat{\mu}(X_i)_{\pi_j} \quad \text{where} \quad Y_i = \pi_j 
+
+
+The quantile :math:`\hat{q}` is then computed the same way as the score method.
```
Comment (LacombeLouis) [inline doc/theoretical_description_classification.rst:57]: Do you mean as in the "LABEL" method?

File: doc/theoretical_description_classification.rst [hunk 3]
```diff
@@ -6,15 +6,179 @@
...
+
+
+.. math:: 
+    \hat{q} = \left \lceil Quantile \left(s_1, ..., s_n ; \frac{\lceil(n+1)(1-\alpha)\rceil}{n}\right) \right\rceil
+
+
+.. math:: 
+   \hat{C}(X_{test}) = \{\pi_1, ..., \pi_{\hat{q}}\} 
+
+
```
Comment (LacombeLouis) [inline doc/theoretical_description_classification.rst:89]: Could be interesting as with the other methods to make a mention of the coverage and possible intuition of this method?

File: doc/theoretical_description_classification.rst [hunk 4]
```diff
@@ -6,15 +6,179 @@
 Theoretical Description
 =======================
 
-TO BE CONTINUED
 
-[2] Mauricio Sadinle, Jing Lei, & Larry Wasserman.
+Three methods for multi-class uncertainty-quantification have been implemented in MAPIE so far :
+LABEL [1], Adaptive Prediction Sets [2, 3] and Top-K [3].
+The difference between these methods is the way the conformity scores are computed. 
+The figure below illustrates the three methods implmented in MAPIE:
```
[the remark this answers is not in the document]
Comment (vtaquet) [reply doc/theoretical_description_classification.rst:13]: Done

File: doc/theoretical_description_classification.rst [hunk 5]
```diff
@@ -6,15 +6,179 @@
...
+    \hat{q} = Quantile \left(s_1, ..., s_n ; \frac{\lceil(n+1)(1-\alpha)\rceil}{n}\right) \\
+
+
+Finally, we construct a prediction set by including all labels with a score higher than the estimated quantile :
+
+.. math:: 
+    \hat{C}(X_{test}) = \{y : \hat{\mu}(X_{test})_y \geq 1 - \hat{q}\}
+
+
+This simple approach allows us to construct prediction sets coming with a theoretical guarantee on the marginal coverage.
```
[the remark this answers is not in the document]
Comment (vtaquet) [reply doc/theoretical_description_classification.rst:40]: Changed to "prediction sets which have a theoretical..."

File: doc/theoretical_description_classification.rst [hunk 6]
```diff
@@ -6,15 +6,179 @@
...
+.. -------------------------------
+
+.. 4. The cross-conformal method
+.. -----------------------------
+
+
+
+.. TO BE CONTINUED
+
+References
```
Comment (LacombeLouis) [inline doc/theoretical_description_classification.rst:171]: Note that references seems to be a sub-part of 4. Split- and cross-conformal methods.

Comment (vtaquet) [reply doc/theoretical_description_classification.rst:171]: Changed.

File: doc/theoretical_description_classification.rst (hunk 3 above)
Comment (vtaquet) [reply doc/theoretical_description_classification.rst:89]: You're right, I added a small paragraph at the beginning of the page to introduce the mathematical setting.

**Pull request merged**